### PR TITLE
Row pinning and search integration

### PR DESCRIPTION
**Pull request by VMihalkov**
Related to #6640

### Additional information (check all that apply):
 - [ ] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by MayaKirova** [comment]
@ChronosSF @VMihalkov  filteredSortedData is not correct when row is pinned initially (for example on afterViewInit) since the pinnedRows collection is not available yet. This can be amended by checking the _pinnedRecordIDs collection instead or using the hasPinnedRecords getter). 
Order is also incorrect if row position is set to RowPinningPosition.Bottom as the pinned rows are always added first in the collection.

**Comment by VMihalkov** [comment]
> @ChronosSF @VMihalkov filteredSortedData is not correct when row is pinned initially (for example on afterViewInit) since the pinnedRows collection is not available yet. This can be amended by checking the _pinnedRecordIDs collection instead or using the hasPinnedRecords getter).
> Order is also incorrect if row position is set to RowPinningPosition.Bottom as the pinned rows are always added first in the collection.

@MayaKirova I tested with RowPinningPosition.Bottom and again the unpinned collection is always going first through the sort pipe, so the order seems to be correct. It works with both Top and Bottom row pinning. I only replaced pinnedRows with _pinnedRecordIDs.

**Comment by MayaKirova** [comment]
@VMihalkov When row is pinned to bottom it actually becomes the last row in the grid, not the first. This is reflected in the row's index. So when findNext is used activation should move to through the unpinned row first, before reaching the pinned row. Currently it moves first to the pinned row, which is incorrect.

For example, if findNext('a') is called and RowPinningPosition.Bottom is used:
image

Should activate "Ana", not 'Alfred'.

File: projects/igniteui-angular/src/lib/grids/grid-base.directive.ts [hunk 1]
```diff
@@ -2935,6 +2937,18 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
         this.setupColumns();
     }
 
+    public setFilteredSortedData(data, pinned: boolean) {
```
Comment (ChronosSF) [inline projects/igniteui-angular/src/lib/grids/grid-base.directive.ts]: this should have @hidden and @Internal otherwise it'll show in the API Docs

**Pull request merged**